pkg/provider/provider.go

```diff
@@ -283,9 +284,14 @@ func (k *kubeProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (
 					Reason: fmt.Sprintf(" Found API Group, but it did not contain a schema for '%s'", gvk),
 				})
 			} else if k8sAPIUnreachable {
+				k8sURL := ""
+				if err, ok := err.(*url.Error); ok {
```

Comment:
Are we confident this is the only time we see this type of error? If not, might make sense to assert `strings.Contains(err.URL, k8sURL)` and that the error message contains `connection refused` -- in which case it seems like would have someway high confidence that this is the error we think it is.

Up to you though.

Reply:
Yeah, I'm reasonably confident in this logic. It's difficult to track down the source of the error, but if we've hit this branch, it should be a `*url.Error`. If not, it just skips printing the URL.